src/tcgwars/logic/impl/gen7/LostThunder.groovy

```diff
@@ -3522,7 +3522,7 @@ public enum LostThunder implements LogicCardInfo {
                 power=false
               }
               before null, null, Source.TRAINER_CARD, {
-                def target=e.getTarget(bg)
+                def target = ef.getResolvedTarget(bg, e)
```

Comment:
not all `ef` instances have this method, so it may throw exception. better add a condition with `ef instanceof TargetedEffect`